clientv3/concurrency/mutex.go

```diff
@@ -55,19 +64,22 @@ func (m *Mutex) TryLock(ctx context.Context) error {
 		m.hdr = resp.Header
 		return nil
 	}
-	client := m.s.Client()
-	// Cannot lock, so delete the key
-	if _, err := client.Delete(ctx, m.myKey); err != nil {
+
+	// Cannot lock so unlock
+	if err := m.unlock(ctx); err != nil {
```

Comment:
this seems strange... why do we unlock when lock fails? probably just keep the old cleanup logic. (i understand the effect the the same, but code repetition here does improve readability)